app/controllers/application_controller.rb

```diff
@@ -18,6 +18,8 @@ class ApplicationController < ActionController::Base
 
   helper :all # include all helpers in the views, all the time
 
+  # set user time zone based on their settings
+  around_action :use_time_zone
```

Comment:
Use `around_action :set_time_zone, if: :current_user` here (to avoid the extra `yield` down in the method).